### PR TITLE
fix: Notify Error Handling Centralization

### PR DESCRIPTION
**Pull request by bryan-robitaille**
# Summary | Résumé

Centralizes the `sendEmail` function calls and error handling into the `@lib/integrations`.
- The error handling includes extracting the exact detailed error message from Notify to include it in the logs.
- Adds a check to ensure the env var `TEMPLATE_ID` is set before calling the `sendEmail` function.

Should provide more context for issue: #3530

### CODE REVIEW

File: lib/integration/notifyConnector.ts [hunk 1]
```diff
@@ -48,3 +49,50 @@ export const getNotifyInstance = () => {
 
   return notifyInstance;
 };
+
+export const sendEmail = async (email: string, personalisation: Record<string, string>) => {
+  try {
+    const templateId = process.env.TEMPLATE_ID;
```
Comment (timarney) [inline lib/integration/notifyConnector.ts:55]: So we allow the options to use a different template ID and default to this one?

Fine with leaving as is just asking questions

File: lib/integration/notifyConnector.ts [hunk 2]
```diff
@@ -48,3 +49,50 @@ export const getNotifyInstance = () => {
...
+  } catch (error) {
+    let errorMessage = "";
+
+    if (error instanceof AxiosError) {
+      if (error.response) {
+        /*
+         * The request was made and the server responded with a
+         * status code that falls out of the range of 2xx
+         */
+        const notifyErrors = Array.isArray(error.response.data.errors)
```
Comment (wmoussa-gc) [inline lib/integration/notifyConnector.ts:71]: do we need to check for presence of  data, errors and status?

File: lib/integration/notifyConnector.ts [hunk 3]
```diff
@@ -48,3 +49,50 @@ export const getNotifyInstance = () => {
...
+         * is an instance of XMLHttpRequest in the browser and an instance
+         * of http.ClientRequest in Node.js
+         */
+        errorMessage = `${error.request}.`;
+      }
+    } else if (error instanceof Error) {
+      errorMessage = `${(error as Error).message}.`;
+    }
+
+    logMessage.error(
```
Comment (wmoussa-gc) [inline lib/integration/notifyConnector.ts:87]: added the future alarm to https://docs.google.com/spreadsheets/d/1FXChe3eJq6BJgMBbF8Y_SXCnoOXoO3IiXMRWt7GoW0w/edit#gid=0

File: lib/integration/notifyConnector.ts (hunk 1 above)
Comment (wmoussa-gc) [reply lib/integration/notifyConnector.ts:55]: I don't think we should include this option unless it's needed at the moment.

File: lib/integration/notifyConnector.ts (hunk 2 above)
Comment (bryan-robitaille) [reply lib/integration/notifyConnector.ts:71]: According to the AxiosError type:
```js
export interface AxiosResponse<T = any, D = any> {
  data: T;
  status: number;
  statusText: string;
  headers: RawAxiosResponseHeaders | AxiosResponseHeaders;
  config: InternalAxiosRequestConfig<D>;
  request?: any;
}
```
`status` and `data` will aways be defined on `response`.
If `errors` is not defined than that `undefined` value will filter through and show up as `"... returned the following detailed errors ."`

**Review by wmoussa-gc — APPROVED** [review]

**Pull request merged**